ci-operator/step-registry/osd-ccs/fleet-manager/sc/provision/osd-ccs-fleet-manager-sc-provision-commands.sh

```diff
@@ -4,28 +4,89 @@ set -o nounset
-#  exit 1
-#fi
-
-
-# Log in
-OCM_VERSION=$(ocm version)
-OCM_TOKEN=$(cat "${CLUSTER_PROFILE_DIR}/ocm-token")
-echo "Logging into ${OCM_LOGIN_ENV} with offline token using ocm cli ${OCM_VERSION}"
-ocm login --url "${OCM_LOGIN_ENV}" --token "${OCM_TOKEN}"
+AWSCRED="${CLUSTER_PROFILE_DIR}/.awscred"
+if [[ -f "${AWSCRED}" ]]; then
+  export AWS_SHARED_CREDENTIALS_FILE="${AWSCRED}"
+  export AWS_DEFAULT_REGION="${OSDFM_REGION}"
+else
+  echo "Did not find compatible cloud provider cluster_profile"
+  exit 1
+fi
 
-# Log in
+# Log in with OSDFM token
 OCM_VERSION=$(ocm version)
-OCM_TOKEN=$(cat "${CLUSTER_PROFILE_DIR}/fleetmanager-token")
+OSDFM_TOKEN=$(cat "${CLUSTER_PROFILE_DIR}/fleetmanager-token")
 echo "Logging into ${OCM_LOGIN_ENV} with offline token using ocm cli ${OCM_VERSION}"
-ocm login --url "${OCM_LOGIN_ENV}" --token "${OCM_TOKEN}"
+if [[ ! -z "${OSDFM_TOKEN}" ]]; then
+  echo "Logging into ${OCM_LOGIN_ENV} with osdfm offline token"
+  ocm login --url "${OCM_LOGIN_ENV}" --token "${OSDFM_TOKEN}"
+  if [ $? -ne 0 ]; then
+    echo "Login failed"
+    exit 1
+  fi
+else
+  echo "Cannot login! You need to specify the offline token OSDFM_TOKEN!"
+  exit 1
+fi
+
+# Check if SC already exists
+sc_cluster_num=$(ocm get /api/osd_fleet_mgmt/v1/service_clusters -p search="region='${OSDFM_REGION}'" | jq -r '.total')
+if ((${sc_cluster_num} > 0)); then
+  sc_cluster_id=$(ocm get /api/osd_fleet_mgmt/v1/service_clusters -p search="region='${OSDFM_REGION}'" | jq -r '.items[0].id')
+  echo "Service Cluster '${sc_cluster_id}' already exists in region '${OSDFM_REGION}', exit"
+  exit 1
+fi
+
+# Create SC
+sc_cluster_id=$(echo '{"region": "'${OSDFM_REGION}'", "cloud_provider": "aws"}' | ocm post /api/osd_fleet_mgmt/v1/service_clusters | jq -r '.id')
+# Save Service Cluster info
+echo "Service Cluster:${sc_cluster_id}"
+echo "${sc_cluster_id}" >> "${SHARED_DIR}/osd-fm-sc-id"
+
+# Check if Service Cluster is ready
+echo "Waiting for Service Cluster ready..."
+wait_for_cluster
+
+# Print Management Cluster info, there might be multiple MC if 1st failed
+mc_cluster_num=$(ocm get /api/osd_fleet_mgmt/v1/management_clusters -p search="parent.id='${sc_cluster_id}'" | jq -r '.total')
+for((i=0;i<${mc_cluster_num};i++));
+do
+  mc_cluster_state=$(ocm get /api/osd_fleet_mgmt/v1/management_clusters -p search="parent.id='${sc_cluster_id}'" | jq -r '.items['$i'].status')
+  echo "Index $i MC is in ${mc_cluster_state} state"
+  # Save 1st ready MC indx to save kubeconfig later
+  if [[ "${mc_cluster_state}" == "ready" ]]; then
+    mc_ready_idx=$i
+  fi
+  mc_cluster_id=$(ocm get /api/osd_fleet_mgmt/v1/management_clusters -p search="parent.id='${sc_cluster_id}'" | jq -r '.items['$i'].id')
+  echo "Management Cluster:${mc_cluster_id}"
+done
+
+#Save MC kubeconfig
+saved_mc_cluster_id=$(ocm get /api/osd_fleet_mgmt/v1/management_clusters -p search="region='${OSDFM_REGION}'" | jq -r '.items['${mc_ready_idx}'].cluster_management_reference.cluster_id')
```

Comment:
Please use `ocm get /api/osd_fleet_mgmt/v1/management_clusters/<mc_id>`.

Besides, `saved_mc_cluster_id` is confused with `mc_cluster_id`, please use `mc_ocm_cluster_id` instead of `saved_mc_cluster_id`. Or using `mc_id` for `mc_cluster_id`, `mc_cluster_id` for `saved_mc_cluster_id` on the ocm side.

Reply:
Yes, they are different Ids actually. I changed it to mc_ocm_cluster_id, thanks